### DISCUSSION TITLE
How to set/add new value to json?

### DISCUSSION THREAD

**Discussion by JHerrKodo** [nlohmann/json#3367] [Q&A] [answered]
How do I add a new key-value-pair, when the key does not exist?
In my case the string "owner/address/street" is passed.

`{
   "id":"1234-5678",
   "owner": {
      "name":"Adam",
      "age": 42
    }
}`

`{
    "id":"1234-5678",
    "owner": {
      "name":"Adam",
      "age": 42,
      "address": {
        "street": "2nd Street"
      }
    }
}`

**Comment by nlohmann** [comment, marked as answer]
Assuming you parsed the JSON value to `j`, then

```cpp
j["owner"]["address"]["street"] = "2nd Street";
```

should do it (not tested).

**Reply by JHerrKodo** [reply]
Thank you for the quick reply. Your code works, but with this solution it is necessary to parse the string.
`j["/owner/address/street"_json_pointer] = "2nd Street"` is also possible.
